src/core/event.js

```diff
@@ -482,9 +482,19 @@ Candy.Core.Event = (function(self, Strophe, $) {
 			}
 
 			if(directInvite.length > 0) {
+				fromUser = Candy.Core.getRoster().get(msg.attr('from'));
```

Comment:
I am inclined to think something API-compatible with a Contact, but with the missing fields returning nil/undefined or empty strings.  May also add a boolean for "in roster?"